workers_controller.js

```diff
@@ -169,9 +169,6 @@ SCWorker.create({
 				scServer.on('connection', socket => {
 					scope.slaveWAMPServer.upgradeToWAMP(socket);
 					socket.on('disconnect', removePeerConnection.bind(null, socket));
-					socket.on('error', err => {
```

Comment:
is socket not connected when it receives error event?

Reply:
It is connected but we don't necessarily need to disconnect it for every error. Some errors like message timeouts will be emitted as an error on the socket but we don't need to disconnect the socket in these cases. The socket will already disconnect itself if there is a fatal error on it.